cli/kata-check_data_ppc64le_test.go

```diff
@@ -112,3 +118,210 @@ model		: 8247-22L
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 48
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 52
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 56
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 60
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 64
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 68
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 72
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 76
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 80
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 84
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 88
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 92
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 96
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 100
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 104
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 108
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 112
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 116
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 120
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+processor	: 124
+cpu		: POWER9, altivec supported
+clock		: 3283.000000MHz
+revision	: 2.2 (pvr 004e 1202)
+
+timebase	: 512000000
+platform	: PowerNV
+model		: 9006-22P
+machine		: PowerNV 9006-22P
+firmware	: OPAL
+MMU		: Radix
+`
+
+const (
+	cpuTypeP8      = 1
+	cpuTypeP9      = 0
+	cpuTypeUnknown = -1
+	archP8         = "POWER8"
+	archP9         = "POWER9"
+)
+
+var testCPUInfoTemplate = setTestCPUInfoTemplate()
+
+func setTestCPUInfoTemplate() string {
+
+	if checkCPUtype() == cpuTypeP8 {
+		return testP8CPUInfoTemplate
+	}
+
+	return testP9CPUInfoTemplate
+
+}
+
+func checkCPUtype() int {
+	var kataLog *logrus.Entry
+
+	content, err := ioutil.ReadFile("/proc/cpuinfo")
+	if err != nil {
+		kataLog.WithError(err).Error("failed to read file")
+		return cpuTypeUnknown
+	}
+	str := string(content)
+	if strings.Contains(str, archP8) {
+		return cpuTypeP8
```

Comment:
I was wondering if it would be more "future proof" to use a regex (something like `cpu\s+:\s*POWER([0-9]+)`) to extract the numeric suffix, convert it to an integer, then just have some code that says something like:

```go
if powerVersion <= 8 {
    // ...
} else {
  // ...
}